### PR TITLE
Use debug package

### PR DESCRIPTION
**Pull request by mshima**
Import and use debug package so `npm test` can be easily debugged.

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/generator-base.js [hunk 1]
```diff
@@ -1484,7 +1490,12 @@ module.exports = class extends PrivateBase {
      * @param {string} msg - message to print
      */
     warning(msg) {
-        this.log(`${chalk.yellow.bold('WARNING!')} ${msg}`);
+        const debug = this.getDebug();
+        if (debug.enabled) {
+            debug(`${chalk.yellow.bold('WARNING!')} ${msg}`);
```
Comment (avdev4j) [inline generators/generator-base.js]: We should factoring this message into a variable and call it twice.

File: generators/generator-base.js [hunk 2]
```diff
@@ -1493,7 +1504,12 @@ module.exports = class extends PrivateBase {
      * @param {string} msg - message to print
      */
     info(msg) {
-        this.log.info(msg);
+        const debug = this.getDebug();
+        if (debug.enabled) {
+            debug(`${chalk.green('INFO!')} ${msg}`);
```
Comment (avdev4j) [inline generators/generator-base.js]: Logging info message into a debug method wrapper is the real behaviour we want ? I don't know the debug package but what should we do that ?

**Comment by avdev4j** [comment]
hi @mshima 
I'm not sure to understand the benefit from the debug package. Could you please provide an explication how to use it or the expected result ?

**Comment by mshima** [comment]
> hi @mshima
> I'm not sure to understand the benefit from the debug package. Could you please provide an explication how to use it or the expected result ?

Debug will be easier on CI and npm test in general.
`DEBUG=namespace:* npm test`

**Comment by mshima** [comment]
I will take a look next week.
I am mobile only this week.

File: generators/generator-base.js (hunk 2 above)
Comment (mshima) [reply generators/generator-base.js]: - when stdin is muted like `npm test`.
- to have blueprint namespace printed with the msg.

**Comment by mshima** [comment]
@avdev4j debug package is already used at yeoman-environment and generator, so it’s already a dependency.

Running `DEBUG=yeoman:* jhipster —with-entities`

The output will be something like this:

```
This is an existing project, using the configuration from your .yo-rc.json file 
to re-generate the project...

2019-10-15T15:21:58.958Z yeoman:generator Running home:vsts:work:1:s:server#askForModuleName
2019-10-15T15:21:58.958Z yeoman:generator Running home:vsts:work:1:s:server#askForServerSideOpts
.
.
2019-10-15T15:21:58.990Z yeoman:generator Running home:vsts:work:1:s:languages#validateFromCli
2019-10-15T15:21:58.990Z yeoman:generator Running home:vsts:work:1:s:languages#setupConsts

Installing languages: en, fr
```

**Comment by mshima** [comment]
@avdev4j _debug variable was added to yeoman-debug 4.5.0.
I removed debug import and cli part and made it very simple.

**Comment by mshima** [comment]
@avdev4j what about this PR.
We should merge it.

**Comment by pascalgrimaud** [comment]
sorry @mshima : can you resolve the conflicts plz ?

**Comment by mshima** [comment]
@pascalgrimaud this PR was included in another one.
I was using it to debug the implementation.

Thank you.

**Comment by mshima** [comment]
@pascalgrimaud I was wrong it wasn't included.

**Pull request merged**